File: Tutorial1-1DaggerBasics/src/main/java/com/example/tutorial1daggerbasics/di/ApplicationModule.java
package com.example.tutorial1daggerbasics.di;


import com.example.tutorial1daggerbasics.model.MyExample;
import com.example.tutorial1daggerbasics.model.MyExampleImpl;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

/**
 * @Singleton annotation in Application level module creates Singleton objects. Without @Singleton
 * new instance of provided object is returned.
 * <p>
 * If provides has a scope then the component of this module should have too
 */
@Module
public class ApplicationModule {
    @Singleton
    @Provides
    MyExample provideMyExample() {
        return new MyExampleImpl();
    }
}

File: Tutorial1-1DaggerBasics/src/main/java/com/example/tutorial1daggerbasics/di/ApplicationModule.kt
package com.example.tutorial1daggerbasics.di

import com.example.tutorial1daggerbasics.model.MyExample
import com.example.tutorial1daggerbasics.model.MyExampleImpl
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

/**
 * @Singleton annotation in Application level module creates Singleton objects. Without @Singleton
 * new instance of provided object is returned.
 *
 * If @Provides method has a scope so should Component has the same scope
 */
@Module
class ApplicationModule {

    @Singleton
    @Provides
    fun provideMyExample(): MyExample {
        return MyExampleImpl()
    }

}